client/components/data/query-site-comments/index.jsx

```diff
@@ -18,8 +18,8 @@ export class QuerySiteComments extends Component {
 		requestComments( this.props );
 	}
 
-	componentDidUpdate( { siteId } ) {
-		if ( siteId !== this.props.siteId ) {
+	componentDidUpdate( { siteId, status } ) {
+		if ( siteId !== this.props.siteId || status !== this.props.status ) {
 			requestComments( this.props );
```

Comment:
I mean calling `requestComments( this.props )` vs `requestComments( { requestSiteStatus, siteId, status } )`. 

The current code is fine in this case since everything is in one file and is short, but I had to scan the file for the function to see what it actually needed. I just wanted to point this case out for future situations.